files/util/index.rb

```diff
@@ -16,6 +17,11 @@ def new_index_file
                   'code-manager': 0,
                   orchestrator:   0, }
       File.write(filepath, tracker.to_yaml)
+      @first_run = true
```

Comment:
So this will be classified as the first run if the index file does not exist? Does the index file get deleted if the module is uninstalled?

Reply:
No, uninstalling the module will not delete this file. Two possible scenarios based on that are also worth considering.

If someone deliberately resets these index numbers to zero without deleting the file, all events will be sent. Also, if someone deletes the file deliberately the module will believe this is the first run.

I'm not sure if either are actually worth doing anything about other than maybe setting a maximum number of events to ever send at once.

Reply:
Manually setting the index to 0 seems like the highest risk from a performance perspective, but idk whether we should just document that or add a max_events guardrail.

Reply:
Summary of offline conversation: We will ticket the handling of the max_events and uninstalling and merge this branch as is.